minichain/state.py

```
@@ -124,22 +186,22 @@ def apply_transaction(self, tx):
+        self.accounts = journal
+
         sender = self.accounts[tx.sender]
         total_cost = tx.amount + (getattr(tx, 'gas_limit', 0) * getattr(tx, 'fee_per_gas', 0))
         
         sender['balance'] -= total_cost
         sender['nonce'] += 1
 
-        import copy
-        state_snapshot = copy.deepcopy(self.accounts)
-
         def rollback_and_refund(error_message, gas_used):
```

Ruff (0.16.0)

[warning] Missing return type annotation for private function `rollback_and_refund`

(ANN202)